### PR TITLE
TGMC: Return of the Pillar

### PR DESCRIPTION
**Pull request by grinkgo**
## About The Pull Request

Ok kwin and i fixed it

## Why It's Good For The Game

lowpop ship whoopee

## Changelog
:cl: grinkgo and Kwin_Original
add: Added a new ship, the TGS Pillar of Spring, for low-mid pop
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by grinkgo** [comment]
> You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

Guess what robot I'm a lead now; you have no power over me!

File: _maps/map_files/Theseus/TGS_Theseus.dmm [hunk 1]
```diff
@@ -2031,14 +2031,6 @@
 	icon_state = "sterile_green_side"
 	},
 /area/mainship/medical/medical_science)
-"afo" = (
-/obj/machinery/door_control/ai/interior{
```
Comment (Rohesie) [inline _maps/map_files/Theseus/TGS_Theseus.dmm]: I assume these changes are unintended?

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -128,3 +128,6 @@
 
 /datum/map_template/shuttle/tgs_canterbury
 	shuttle_id = "tgs_canterbury"
+
+/datum/map_template/shuttle/escape_shuttle
+	shuttle_id = "escape_shuttle"
```
Comment (Rohesie) [inline code/datums/shuttles.dm]:
```suggestion
	shuttle_id = "escape_shuttle"

```
Try to leave a blank line at the end of a file.

File: code/game/objects/machinery/door_control.dm [hunk 3]
```diff
@@ -158,3 +158,77 @@
...
+	id = "ROlobby"
+	req_one_access = list(ACCESS_MARINE_CARGO, ACCESS_MARINE_CAPTAIN)
+
+/obj/machinery/door_control/mainship/req/ro1	
+	name = "RO Line 1 Shutters"
+	id = "ROlobby1"
+
+/obj/machinery/door_control/mainship/req/ro2	
+	name = "RO Line 2 Shutters"
+	id = "ROlobby2"
```
Comment (Rohesie) [inline code/game/objects/machinery/door_control.dm]:
```suggestion
	id = "ROlobby2"

```
Same here.

File: tgmc.dme [hunk 4]
```diff
@@ -12,6 +12,7 @@
 // BEGIN_INCLUDE
 #include "_maps\_basemap.dm"
 #include "_maps\map_files\generic\Admin_Level.dmm"
+#include "_maps\map_files\Pillar_of_Spring\TGS_Pillar_of_Spring.dmm"
```
Comment (psykzz) [inline tgmc.dme]: like the Theseus this shouldn't be included

File: _maps/map_files/Theseus/TGS_Theseus.dmm (hunk 1 above)
Comment (grinkgo) [reply _maps/map_files/Theseus/TGS_Theseus.dmm]: Uh, I'm not sure actually. But that typepath is outdated, no?

File: tgmc.dme (hunk 4 above)
Comment (grinkgo) [reply tgmc.dme]: Ok i took it out

**Comment by grinkgo** [comment]
> avoid changing current maps when adding a new map.
> 
> Also add it to _maps/_basemap.dm

Travis ree'd at me about multiple areas on the Theseus :<

**Comment by spookydonut** [comment]
> 
> 
> > avoid changing current maps when adding a new map.
> > Also add it to _maps/_basemap.dm
> 
> Travis ree'd at me about multiple areas on the Theseus :<

Well what typepaths are you changing?  Do them in a different PR then you wont end up in constant conflict because of Theseus changes :P

File: _maps/map_files/Theseus/TGS_Theseus.dmm (hunk 1 above)
Comment (Rohesie) [reply _maps/map_files/Theseus/TGS_Theseus.dmm]: Only in your PR.
Could you do this on a different PR, like spooky suggested?

**Comment by grinkgo** [comment]
> > > avoid changing current maps when adding a new map.
> > > Also add it to _maps/_basemap.dm
> > 
> > 
> > Travis ree'd at me about multiple areas on the Theseus :<
> 
> Well what typepaths are you changing? Do them in a different PR then you wont end up in constant conflict because of Theseus changes :P

Hm, I actually don't know why there were Theseus changes in that commit. It adds areas from desert dam? And also it updates the typepath for the AI doors but if anything that would've come up in an earlier commit

File: _maps/map_files/Theseus/TGS_Theseus.dmm (hunk 1 above)
Comment (grinkgo) [reply _maps/map_files/Theseus/TGS_Theseus.dmm]: How do I not include theseus in my pr

File: _maps/map_files/Theseus/TGS_Theseus.dmm [hunk 5]
```diff
@@ -86061,7 +86061,7 @@ hKL
 aoH
 apS
 awK
-afo
+pvn
```
Comment (Rohesie) [inline _maps/map_files/Theseus/TGS_Theseus.dmm]:
```suggestion
afo
```

File: _maps/map_files/Theseus/TGS_Theseus.dmm [hunk 6]
```diff
@@ -86575,7 +86575,7 @@ aoH
 aoH
 apS
 vVX
-ayB
+pvn
```
Comment (Rohesie) [inline _maps/map_files/Theseus/TGS_Theseus.dmm]:
```suggestion
ayB
```

File: code/game/objects/machinery/doors/shutters.dm [hunk 7]
```diff
@@ -147,3 +147,53 @@
...
+	name = "\improper CIC Shutters"	
+
+/obj/machinery/door/poddoor/shutters/mainship/cic/armory
+	name = "\improper Armory Shutters"
+	id = "cic_armory"
+	icon_state = "shutter1"
+
+/obj/machinery/door/poddoor/shutters/mainship/corporate
+	name = "\improper Privacy Shutters"
+	id = "cl_shutters"
```
Comment (Rohesie) [inline code/game/objects/machinery/doors/shutters.dm]:
```suggestion
	id = "cl_shutters"

```
Have files end in an empty line whenever possible.

File: _maps/Pillar_of_Spring.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+    "map_name": "Pillar of Spring",
+    "map_path": "map_files/Pillar_of_SPring",
+    "map_file": "TGS_Pillar_of_Spring.dmm",
+	"traits": [{"Marine Main Ship": true}], 
+    "shuttles": {
+    }
+}
```
Comment (Rohesie) [inline _maps/Pillar_of_Spring.json]:
```suggestion
}

```
Another little line at the end of the file. Not as relevant.

File: _maps/map_files/Theseus/TGS_Theseus.dmm (hunk 5 above)
Comment (Rohesie) [inline _maps/map_files/Theseus/TGS_Theseus.dmm]:
```suggestion
afo
```

File: _maps/map_files/Theseus/TGS_Theseus.dmm (hunk 6 above)
Comment (Rohesie) [inline _maps/map_files/Theseus/TGS_Theseus.dmm]:
```suggestion
ayB
```

**Comment by Rohesie** [comment]
Code-wise this seems fine now.
Is it tested locally?

**Comment by grinkgo** [comment]
> Code-wise this seems fine now.
> Is it tested locally?

ayup

**Comment by Mark9013100** [comment]
Suggestion: Move the supply ASRS console next to the elevator.
Map Error: The escape shuttle cannot be entered due to a wall placed under the door.

**Comment by grinkgo** [comment]
> Suggestion: Move the supply ASRS console next to the elevator.
> Map Error: The escape shuttle cannot be entered due to a wall placed under the door.

Suggestion denied.
Map error is resolved, I'll double check that the server updates it properly

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**